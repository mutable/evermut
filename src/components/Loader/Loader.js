import React from 'react';
import PropTypes from 'prop-types';
import { Pane, Spinner } from 'evergreen-ui';

const Loader = ({ style, height, ...rest }) => {
  const props = Object.keys(style).length ? { theme: style } : {};

  return (
    <Pane
      display="flex"
      alignItems="center"
      justifyContent="center"
      flex={1}
      height={height}
      width='100%'
      {...rest}
    >
      <Spinner {...props} />
    </Pane>
  )
}

Loader.propTypes = {
  style: PropTypes.object,
  height: PropTypes.string
}

Loader.defaultProps = {
  style: {},
  height: '100vh'
}

export default Loader;
