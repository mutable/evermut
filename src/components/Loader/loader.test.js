import React from 'react';
import renderer from 'react-test-renderer';

import Loader from '../Loader';

describe('<Loader />', () => {
  it('component render correctly', () => {
    const component = renderer.create(<Loader />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('with the changed height of wrapper', () => {
    const loader = renderer.create(<Loader height="100px" />);
    const tree = loader.toJSON();
    expect(tree).toMatchSnapshot();
  });
});