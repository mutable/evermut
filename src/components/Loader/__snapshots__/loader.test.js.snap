// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> component render correctly 1`] = `
<div
  className="css-nil ðŸ“¦dspl_flex ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦flx_1 ðŸ“¦h_100vh ðŸ“¦w_100prcnt ðŸ“¦box-szg_border-box"
>
  <div
    className="ðŸ“¦w_40px ðŸ“¦h_40px ðŸ“¦ln-ht_0 ðŸ“¦box-szg_border-box"
    delay={0}
  >
    <svg
      className="css-11r1ktn ðŸ“¦box-szg_border-box"
      viewBox="0 0 150 150"
      x="0px"
      y="0px"
    >
      <circle
        className="css-1ipoqbj ðŸ“¦box-szg_border-box"
        cx="75"
        cy="75"
        r="60"
      />
    </svg>
  </div>
</div>
`;

exports[`<Loader /> with the changed height of wrapper 1`] = `
<div
  className="css-nil ðŸ“¦dspl_flex ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦flx_1 ðŸ“¦h_100px ðŸ“¦w_100prcnt ðŸ“¦box-szg_border-box"
>
  <div
    className="ðŸ“¦w_40px ðŸ“¦h_40px ðŸ“¦ln-ht_0 ðŸ“¦box-szg_border-box"
    delay={0}
  >
    <svg
      className="css-11r1ktn ðŸ“¦box-szg_border-box"
      viewBox="0 0 150 150"
      x="0px"
      y="0px"
    >
      <circle
        className="css-1ipoqbj ðŸ“¦box-szg_border-box"
        cx="75"
        cy="75"
        r="60"
      />
    </svg>
  </div>
</div>
`;
